src/main/java/com/github/maleksandrowicz93/cqrsdemo/student/exception/InvalidCredentialsException.java

```diff
@@ -0,0 +1,10 @@
+package com.github.maleksandrowicz93.cqrsdemo.student.exception;
+
+public class InvalidCredentialsException extends StudentException {
+
+    private static final ErrorMessage ERROR_MESSAGE = ErrorMessage.INVALID_CREDENTIALS;
```

Comment:
zbędna stała, to już jest enum, spokojnie może być inline w konstruktorze - nie ma z tego żadnej korzyści a nieco zmniejsza czytelność

Reply:
ta sama uwaga w pozostałych analogicznych miejscach

Reply:
Stała utworzona na wypadek przyszłego dodania kilku konstruktorów. Po co? Np obsługa i18n z substitution map do wstrzykiwania wartości w placeholdery tłumaczonego tekstu, który musi być customizowany jakąś nazwą, czy id. Wtedy miałbym właśnie 2 konstruktory - przyjmujący tylko ErrorMessage i inicjalizujący pustą mapę, oraz przyjmujący oba parametry. I stała robi się przydatna :) Pytanie, czy w takiej sytuacji definiowanie stałej z myślą o przyszłości to dobry pattern, czy nie. Z tyłu głowy mam dodanie potem i18n, nie dodałem chyba tego w roadmapie projektu :(

Reply:
okej, ale dlaczego przepisywać stałą wartość enuma do innej stałej w klasie, zamiast od razu wykorzystać go w konstruktorze, bo chyba tego nie czuję?

Reply:
Nie lubię wpisywać w kilku miejscach tej samej wartości, wolę mieć stałą. Wolę w kilku konstruktorach użyć stałej i wszędzie wpisać ERROR_MESSAGE, niż x razy wklepywać ErrorMessage.TEN_LUB_TAMTEN_MESSAGE. I wtedy mam większa szansę, że się nie pomylę xD

Reply:
Ach, rozumiem teraz, to a propos

> czy w takiej sytuacji definiowanie stałej z myślą o przyszłości to dobry pattern, czy nie

nie :p Komplikuje i rodzi pytania bez potrzeby + YAGNI ;)

Reply:
w sumie fakt, YAGNI nie musi się przecież ograniczać do nieużywanego kodu, konstrukcje używane, ale "na kiedyś" też przecież mogą się w to wpisywać, chyba mnie to przekonało :) teraz więc usunę stałe, ale dodam, gdy powstanie kolejny konstruktor 😎